misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java

```diff
@@ -406,41 +411,53 @@ private void executeFunction(Test test, TestSuite suite, String packageName, Cla
         }
     }
 
-    private void executeAfterFunction(Test test, TestSuite suite, ClassLoader classLoader, Scheduler scheduler)  {
-        try {
-            if (test.getAfterTestFunction() != null) {
-                invokeTestFunction(suite, test.getAfterTestFunction(), classLoader, scheduler);
+    private void executeAfterFunction(Test test, TestSuite suite, ClassLoader classLoader, Scheduler scheduler,
+                                      AtomicBoolean shouldSkip, AtomicBoolean shouldSkipTest,
+                                      List<String> failedAfterFuncTests)  {
+        if (!shouldSkip.get() && !shouldSkipTest.get()) {
+            try {
+                if (test.getAfterTestFunction() != null) {
+                    invokeTestFunction(suite, test.getAfterTestFunction(), classLoader, scheduler);
+                }
+            } catch (Throwable e) {
+                failedAfterFuncTests.add(test.getTestName());
+                String error = String.format("\t[fail] " + test + " [after test function for the test %s] :\n\t    %s",
+                        test, formatErrorMessage(e));
+                errStream.println(error);
             }
-        } catch (Throwable e) {
-            String error = String.format("\t[fail] " + test + " [after test function for the test %s] :\n\t    %s",
-                                  test, formatErrorMessage(e));
-            errStream.println(error);
         }
     }
 
-    private void executeAfterEachFunction(Test test, TestSuite suite, ClassLoader classLoader, Scheduler scheduler) {
-        suite.getAfterEachFunctionNames().forEach(afterEachTest -> {
-            String errorMsg2;
-            try {
-                invokeTestFunction(suite, afterEachTest, classLoader, scheduler);
-            } catch (Throwable e) {
-                errorMsg2 = String.format("\t[fail] " + afterEachTest +
-                                                  " [after each test function for the test %s] :\n\t    %s",
-                                          test, formatErrorMessage(e));
-                errStream.println(errorMsg2);
-            }
-        });
+    private void executeAfterEachFunction(Test test, TestSuite suite, ClassLoader classLoader, Scheduler scheduler,
+                                          AtomicBoolean shouldSkip, AtomicBoolean shouldSkipTest) {
+        if (!shouldSkip.get() && !shouldSkipTest.get()) {
+            suite.getAfterEachFunctionNames().forEach(afterEachTest -> {
+                String errorMsg2;
```

Comment:
Shall we have a more meaningful variable name here?

Reply:
Fixed in f9d41bc